mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapperTest.java

```diff
@@ -0,0 +1,189 @@
+import java.util.concurrent.Executors;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.modules.CassandraAnnotationModule;
+import org.apache.james.mailbox.cassandra.table.CassandraAnnotationTable;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxAnnotation.AnnotationValue;
+import org.apache.james.mailbox.model.MailboxAnnotation.MailboxAnnotationEntryKey;
+import org.apache.james.mailbox.model.MailboxAnnotation.MailboxAnnotationEntryValue;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.SimpleMailboxAnnotation;
+import org.apache.james.mailbox.model.SimpleMailboxAnnotation.SimpleMailboxAnnotationCommand;
+import org.apache.james.mailbox.model.SimpleMailboxAnnotation.SimpleMailboxAnnotationEntryKey;
+import org.apache.james.mailbox.model.SimpleMailboxAnnotation.SimpleMailboxAnnotationEntryValue;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datastax.driver.core.Session;
+
+public class CassandraAnnotationMapperTest {
+    private CassandraAnnotationMapper cassandraAnnotationMapper;
+    private CassandraCluster cassandra;
+    private SimpleMailbox<CassandraId> mailbox;
+    private int uidValidity;
+    private int maxRetry;
+    private ExecutorService executor;
+
+    @Before
+    public void setUp() throws Exception {
+        cassandra = CassandraCluster.create(new CassandraAnnotationModule());
+        cassandra.ensureAllTables();
+        uidValidity = 10;
+        mailbox = new SimpleMailbox<>(new MailboxPath("#private", "benwa@linagora.com", "INBOX"), uidValidity);
+        mailbox.setMailboxId(CassandraId.of(UUID.fromString("464765a0-e4e7-11e4-aba4-710c1de3782b")));
+        maxRetry = 100;
+        cassandraAnnotationMapper = new CassandraAnnotationMapper(mailbox, cassandra.getConf(), maxRetry);
+        executor = Executors.newFixedThreadPool(2);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cassandra.clearAllTables();
+        executor.shutdownNow();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void creatingAnnotaionMapperWithNegativeMaxRetryShouldFail() {
+        new CassandraAnnotationMapper(mailbox, cassandra.getConf(), -1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void creatingAnnotationMapperWithNullMaxRetryShouldFail() {
+        new CassandraAnnotationMapper(mailbox, cassandra.getConf(), 0);
+    }
+
+    @Test
+    public void retrieveEmptyAnnotation() throws Exception {
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(SimpleMailboxAnnotation.EMPTY);
+    }
+
+    @Test
+    public void retrieveAnnotation() throws Exception {
+        cassandra.getConf().execute(
```

Comment:
we usually don't access the database directly from test (it's actually a recurring debate in the team but we settled for 'test everything from API point of view')